src/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -101,10 +107,39 @@ public ClusterDataToSamConverter(final String runBarcode,
     }
 
     /**
+     *  Sets the SAM tag to use to store the molecular index bases.  If multiple molecular indexes exist, it will concatenate them
+     * and store them in this tag.
+     */
+    public ClusterDataToSamConverter withMolecularIndexTag(final String molecularIndexTag) {
```

Comment:
I'm intregued by the use of the word "with" in the name of this method, when the verb is clearly "set"...why not the set* pattern we have in many other factories?